src/main/java/com/ausregistry/jtoolkit2/se/DomainAddRem.java

```diff
@@ -47,16 +49,48 @@ public DomainAddRem(AddRemType type, String[] nameservers,
         }
     }
 
+    /**
+     * Maximal specification of the attribute values which may be added or
+     * removed from a domain. Each of the parameters is optional, but at least
+     * one must be specified.
+     */
+    public DomainAddRem(AddRemType type, Host[] nameservers,
+        String[] techContacts, String[] adminContacts,
+        String[] billingContacts, Status[] statuses) {
+        this.type = type.toString();
+
+        if (nameservers != null) {
+            this.hosts = nameservers.clone();
+        }
+        if (techContacts != null) {
+            this.techContacts = techContacts.clone();
+        }
+        if (adminContacts != null) {
+            this.adminContacts = adminContacts.clone();
+        }
+        if (billingContacts != null) {
+            this.billingContacts = billingContacts.clone();
+        }
+        if (statuses != null) {
+            this.statuses = statuses.clone();
+        }
+    }
+
     public Element appendToElement(XMLWriter xmlWriter, Element parent) {
         Element addRem = xmlWriter.appendChild(parent, type);
 
         if (nameservers != null) {
             Element ns = xmlWriter.appendChild(addRem, "ns");
-
             for (String hostObj : nameservers) {
                 xmlWriter.appendChild(ns, "hostObj").setTextContent(hostObj);
             }
         }
+        if (hosts != null) {
```

Comment:
With this implementation, there are two fields in the class, one is a `String[] nameservers`, the other is a `Host[] hosts`. Depends on which is not null the code decides how to render it. 

It would be better to have only one field, say to keep `Host[] hosts`, and add a boolean field to express the intention regarding how to render the XML.